File: src/main/java/uk/co/alt236/apkcompare/comparators/ResultBlock.java
package uk.co.alt236.apkcompare.comparators;


import java.util.List;

public class ResultBlock implements ComparisonResult {

    private final List<ResultItem> resultItems;
    private final String title;

    public ResultBlock(String title, List<ResultItem> resultItems) {
        this.resultItems = resultItems;
        this.title = title;
    }

    public List<ResultItem> getResultItems() {
        return resultItems;
    }

    @Override
    public String getTitle() {
        return title;
    }

    @Override
    public Similarity getSimilarity() {
        return CollectionSimilarityEvaluator.evaluate(resultItems);
    }

}

File: app/src/main/java/uk/co/alt236/apkcompare/app/comparators/results/groups/ResultBlock.kt
package uk.co.alt236.apkcompare.app.comparators.results.groups

import uk.co.alt236.apkcompare.app.comparators.results.ComparisonResult
import uk.co.alt236.apkcompare.app.comparators.results.EnumEvaluators
import uk.co.alt236.apkcompare.app.comparators.results.MissingValue
import uk.co.alt236.apkcompare.app.comparators.results.Similarity

data class ResultBlock @JvmOverloads constructor(override val title: String,
                                                 val comparisonResults: List<ComparisonResult>,
                                                 val complex: Boolean = false) : ComparisonResult {

    override val similarity: Similarity by lazy { EnumEvaluators.evaluateSimilarity(comparisonResults) }
    override val missingValue: MissingValue by lazy { EnumEvaluators.evaluateMissingValue(comparisonResults) }
}
